base/logging.go

```diff
@@ -804,25 +804,21 @@ var (
 )
 
 // RotateLogfiles rotates all active log files.
-func RotateLogfiles() error {
+func RotateLogfiles() map[*FileLogger]error {
 	Infof(KeyAll, "Rotating log files...")
 
-	// Even if one errors, we still want to try and rotate the rest, hence the weird error handling.
-	var err error
-	if e := debugLogger.Rotate(); e != nil {
-		err = e
+	loggers := map[*FileLogger]error{
```

Comment:
I think this could just be initialized to an empty map

Reply:
I need something to iterate over, there's no iterable list of file loggers anywhere else unfortunately.